scripts/readelf.py

```diff
@@ -815,15 +815,14 @@ def display_hex_dump(self, section_spec):
             sys.stderr.write('readelf: Warning: Section \'%s\' was not dumped because it does not exist!\n' % (
                 section_spec))
             return
-        if section['sh_type'] == 'SHT_NOBITS':
-            self._emitline("\nSection '%s' has no data to dump." % (
+        if section['sh_type'] == 'SHT_NOBITS' or not (data := section.data()):
```

Comment:
Maybe I'm missing something, but what's the point of `data := ...` here? You're not using `data` inside the condition body

Reply:
I moved the `data = section.data()` here from 3 lines below: Otherwise I would have to either repeat the output:
```python
if section['sh_type'] == 'SHT_NOBITS':
    self._emitline("\nSection '%s' has no data to dump." % …)
data = section.data()
if not data:
    self._emitline("\nSection '%s' has no data to dump." % …)
```
or *always* call `section.data()` even in the case where `sh_type = SHT_NOBITS`:
```python
data = section.data()
if section['sh_type'] == 'SHT_NOBITS' or not data:
    self._emitline("\nSection '%s' has no data to dump." % …)
```
By using the walrus operator I get both: no repetition and the lazy calling. YMMV